modules/calcite/src/main/java/org/apache/ignite/internal/processors/query/calcite/schema/IgniteTableImpl.java

```diff
@@ -59,16 +62,21 @@
     /** */
     private final Statistic statistic;
 
+    /** */
+    private final GridKernalContext ctx;
+
     /** */
     private volatile GridH2Table tbl;
 
     /** */
     private final Map<String, IgniteIndex> indexes = new ConcurrentHashMap<>();
 
     /**
-     *  @param desc Table descriptor.
+     * @param ctx Kernal context.
+     * @param desc Table descriptor.
      */
-    public IgniteTableImpl(TableDescriptor desc) {
+    public IgniteTableImpl(GridKernalContext ctx, TableDescriptor desc) {
```

Comment:
Let's avoid use KernalContext

Reply:
But why? IgniteTableImpl already uses KernelContext, but obtains it from cache context (which is not always initialized). Also, there is no other way to get GridCacheProcessor to start lazy caches.

Reply:
GridCacheProcessor and GridQueryProcessor required, not whole KernalContext, isn't it?
I understand that it widely used practice in Ignite when we pass CTX everywhere. However, it makes code more coupled. Most of the code will be moved into Ignite 3.0, let's try to make it better from now.

Reply:
I think if at least two components are required it's better to pass only the KernalContext. Passing two components and store them in two fields only brings unnecessary code complexity.
I can also pass and store only one of these components and get another from it (for example `cacheProc.context().kernelContext().query()`) but it looks weird.